sympy/polys/rings.py

```diff
@@ -2958,6 +2960,279 @@ def subresultants(self, g, x=None):
 
         return R
 
+    def main_variable(self):
+        """
+        Find the smallest index of the free symbolic variable in a given
+        polynomial.
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ, ring
+        >>> R, x, y = ring("x, y", ZZ)
+        >>> p = y**2 + 2 - y**3 + 4*y
+        >>> p.main_variable()
+        1
+
+        """
+        p = self
+        exponents = list(map(sum, zip(*p)))
+        free_sym = {n for n, e in enumerate(exponents) if e}
+        return min(free_sym)
+
+    def monomial_extract(p):
+        """
+        Extracts any common monomial from the polynomials in p.
```

Comment:
There seems to be some confusion here about whether this is a method or a function. If `p` is supposed to be a list of polynomials then this won't work as a method. Probably it should just be a function defined after the class statement.